raiden-ts/src/transport/epics.ts

```diff
@@ -83,6 +92,47 @@ import { getPresences$, getRoom$, roomMatch, globalRoomNames } from './utils';
 const AVAILABLE = ['online', 'unavailable'];
 const userRe = /^@(0x[0-9a-f]{40})[.:]/i;
 
+/**
+ * Creates and returns a matrix filter. The filter reduces the size of the initial sync by
+ * filtering out broadcast rooms, emphemeral messages like receipts etc.
+ *
+ * @param config - The {@link RaidenConfig} provides the broadcast room aliases for pfs and discovery.
+ * @param server - The matrix server used e.g., https://transport04.raiden.network is part of the alias.
+ * @param matrix - The {@link MatrixClient} instance used to create the filter.
+ */
+async function createFilter(config: RaidenConfig, server: string, matrix: MatrixClient) {
+  const { pfsRoom, discoveryRoom } = config;
+  const transport = server.split('://')[1];
+
+  const filteredRooms = [];
+
+  if (pfsRoom) {
+    const { room_id: roomId } = await matrix.getRoomIdForAlias(`#${pfsRoom}:${transport}`);
```

Comment:
Each `getRoomIdForAlias` is an http request, which is slow. Both should be done in parallel, when needed. Besides this, I believe the global rooms's `joinRoom` returns roomId somehow, so you could use it to save a request for each, although it may require putting global joinRoom logic here as well